### DISCUSSION TITLE
After upgrading Longhorn, multiple instance-managers exist

### DISCUSSION THREAD

**Discussion by githubeto** [longhorn/longhorn#8702] [Q&A]
We have upgraded Longhorn from v1.4.1 to v1.5.5 to v1.6.1, and there are multiple instance-manager-xxx mixed in. According to the [documentation here](https://longhorn.io/kb/troubleshooting-some-old-instance-manager-pods-are-still-running-after-upgrade/), this seems to be an expected situation, but we would like to unify them to the latest version.

As [mentioned ](https://github.com/longhorn/longhorn/issues/2144#issuecomment-754416093
)in this comment, Does this mean that simply restarting (rolling out) the Pods on the application side using the PVC will update the instance-managers to the latest version?

Also, there is one more thing I don't understand about the Volume. When looking at the Volume details screen, the left menu lists the engine image from v1.4.1 as the Instance Manager, but each Replica lists the instance-manager type "aio" from the upgraded v1.6.1 as the Instance Manager Image. What does this mean? Which one is correct?

![2024-06-05_10h25_52](https://github.com/longhorn/longhorn/assets/122009205/f34bc788-f3bd-41d5-95dc-901856374b93)

**Comment by githubeto** [comment]
Rolling out the Deployment of the Pod for the application using PVC did not change anything.

**Reply by mantissahz** [reply]
Did you upgrade the engine of the volumes? ([document here, a step of the upgrade](https://longhorn.io/docs/1.6.2/deploy/upgrade/upgrade-engine/))
At 1.4.x, engine pods and instance pods are separated.
You need to upgrade the engine first and scale down the workload to detach the volume. And then scale up the workload to attach the volume again.

**Reply by githubeto** [reply]
Thank you for your comment.
I hadn't found any clear documentation on this. I understand now.
By the way, when you refer to "engine pods," do you mean instance-manager-e-xxxx, and when you refer to "instance pods," do you mean instance-manager-r-xxxx?

**Reply by mantissahz** [reply]
>By the way, when you refer to "engine pods," do you mean instance-manager-e-xxxx, and when you refer to "instance pods," do you mean instance-manager-r-xxxx?

Yes, that's what I meant. Now they are combined into the `instance-manager-xxxxx` pods.

**Reply by githubeto** [reply]
Sure, I understand. Thank you.
